gateway/src/main/kotlin/ConnectionManagedGateway.kt

```diff
@@ -0,0 +1,128 @@
+package dev.kord.gateway
+
+import dev.kord.common.entity.optional.optional
+import dev.kord.common.entity.optional.optionalInt
+import dev.kord.gateway.connection.GatewayConnection
+import dev.kord.gateway.connection.GatewayConnection.CloseReason
+import dev.kord.gateway.connection.GatewayConnection.Session
+import kotlinx.atomicfu.AtomicRef
+import kotlinx.atomicfu.atomic
+import kotlinx.atomicfu.loop
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.serialization.json.Json
+import java.net.URI
+import kotlin.coroutines.CoroutineContext
+import kotlin.time.Duration
+
+internal class ConnectionManagedGateway(
+    private val connectionProvider: suspend () -> GatewayConnection,
```

Comment:
I don't like to see such declarations in code as it doesn't really give a hint on what it is used for outside of this context.
Provide a typealias for it please.